test/test_biogeochemistry.jl

```diff
@@ -0,0 +1,308 @@
+                                            mortality_rate = 0.1/day,
+                                            surface_PAR = t -> 100*max(0.0, sin(t*π/(12hours))),
+                                            sinking_velocity = 200/day
+                                            )
+
+model = NonhydrostaticModel(; grid, biogeochemistry,
+                            advection = WENO(; grid),
+                            timestepper = :RungeKutta3,
+                            closure = ScalarDiffusivity(ν=1e-4, κ=1e-4),
+                            coriolis = FPlane(f=1e-4),
+                            tracers = :b,
+                            buoyancy = BuoyancyTracer(),
+                            boundary_conditions = (; b=buoyancy_bcs))
+
+mixed_layer_depth = 32 # m
+stratification(z) = z < -mixed_layer_depth ? N² * z : - N² * mixed_layer_depth
+noise(z) = 1e-4 * N² * grid.Lz * randn() * exp(z / 4)
+initial_buoyancy(x, y, z) = stratification(z) + noise(z)
+
+set!(model, b=initial_buoyancy, P = 1.0)
+
+simulation = Simulation(model, Δt=2minutes, stop_time=1day)
+
+wizard = TimeStepWizard(cfl=1.0, max_change=1.1, max_Δt=2minutes)
+simulation.callbacks[:wizard] = Callback(wizard, IterationInterval(10))
+
+progress(sim) = @printf("Iteration: %d, time: %s, Δt: %s\n",
+                        iteration(sim), prettytime(sim), prettytime(sim.Δt))
+
+simulation.callbacks[:progress] = Callback(progress, IterationInterval(100))
+
+outputs = (w = model.velocities.w,
+           P = model.tracers.P,
+           PAR = model.auxiliary_fields.PAR,
+           avg_P = Average(model.tracers.P, dims=(1, 2)))
+
+simulation.output_writers[:simple_output] =
+    JLD2OutputWriter(model, outputs,
+                     schedule = TimeInterval(20minutes),
+                     filename = "biogeochemistry_test.jld2",
+                     overwrite_existing = true)
+
+run!(simulation)
+=#
+#####
+##### Example using SomethingBiogeochemistry
+#####
+
+@inline growth(x, y, z, t, P, PAR, μ₀, k, m, other_params...) = (μ₀ * (1 - exp(-PAR/k)) - m) * P
+
+@inline function integrate_light!(bgc, model)
+    # have to slightly rewrite to pass parameters
+    par_calculation =  launch!(model.architecture, model.grid, :xy, update_PAR!,
+                               model.auxiliary_fields.PAR, model.grid, model.tracers.P, model.clock.time, bgc.parameters,
+                               dependencies = Event(device(model.architecture)))
+    wait(device(model.architecture), par_calculation)
+end
+
+biogeochemistry_parameters = (
+    growth_rate = 1/day,
+    light_limit = 3.5,
+    mortality_rate = 0.1/day,
+
+    water_light_attenuation_coefficient = 0.12,
+    phytoplankton_light_attenuation_coefficient = 0.06,
+    phytoplankton_light_attenuation_exponent = 0.6,
+    surface_PAR = t -> 100*max(0.0, sin(t*π/(12hours)))
+)
+
+biogeochemistry = SomethingBiogeochemistry(tracers = :P, 
+                                           auxiliary_fields = :PAR, 
```

Comment:
Should be this eventually:

```suggestion
                                           auxiliary_fields = (; PAR=PARField(grid)), 
```